services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java

```diff
@@ -16,30 +16,125 @@
 package org.openkilda.wfm.topology.nbworker.bolts;
 
 import org.openkilda.messaging.info.InfoData;
+import org.openkilda.messaging.info.event.IslInfoData;
+import org.openkilda.messaging.info.event.PathInfoData;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.messaging.info.flow.FlowsResponse;
+import org.openkilda.messaging.model.SwitchId;
 import org.openkilda.messaging.nbtopology.request.BaseRequest;
+import org.openkilda.messaging.nbtopology.request.GetFlowIdsForLinkRequest;
 import org.openkilda.pce.provider.Auth;
+import org.openkilda.wfm.topology.nbworker.converters.LinksConverter;
 
 import org.apache.storm.topology.OutputFieldsDeclarer;
 import org.apache.storm.tuple.Fields;
 import org.apache.storm.tuple.Tuple;
 import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.neo4j.driver.v1.Value;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class FlowOperationsBolt extends NeoOperationsBolt {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(FlowOperationsBolt.class);
+    private static final Logger logger = LoggerFactory.getLogger(FlowOperationsBolt.class);
 
     public FlowOperationsBolt(Auth neoAuth) {
         super(neoAuth);
     }
 
     @Override
     List<? extends InfoData> processRequest(Tuple tuple, BaseRequest request, Session session) {
-        LOGGER.warn("Received unsupported Flow Operation");
-        return null;
+        List<? extends InfoData> result = null;
+        if (request instanceof GetFlowIdsForLinkRequest) {
+            result = getFlowIdsForLink((GetFlowIdsForLinkRequest) request, session);
+        } else {
+            unhandledInput(tuple);
+        }
+
+        return result;
+    }
+
+    private List<FlowsResponse> getFlowIdsForLink(GetFlowIdsForLinkRequest request, Session session) {
```

Comment:
You do not match to design https://github.com/telstra/open-kilda/tree/develop/docs/design/storm-decouple